File: code/community/1301705198959333384/reduce-latency-deepgram-zoom.js
const Deepgram = require('@deepgram/sdk');
const WebSocket = require('ws');

// Set your Deepgram API key and audio source
const DEEPGRAM_API_KEY = process.env.DEEPGRAM_API_KEY;
const AUDIO_SOURCE = 'your-zoom-websocket-url';

// Initialize the Deepgram SDK
const deepgram = new Deepgram(DEEPGRAM_API_KEY);

// Configure WebSocket to receive audio stream
const ws = new WebSocket(deepgram.transcription.live({
  url: AUDIO_SOURCE,
  config: {
    punctuate: true,
    interimResults: true
  }
}));

// Handle incoming messages
ws.on('message', (message) => {
  const parsedMessage = JSON.parse(message);
  if (parsedMessage.is_final) {
    console.log(`Transcript: ${parsedMessage.channel.alternatives[0].transcript}`);
  }
});

// Handle WebSocket errors
ws.on('error', (error) => {
  console.error(`WebSocket error: ${error}`);
});

// Close WebSocket connection
process.on('SIGINT', () => {
  ws.close();
  process.exit();
});

// Links to further resources, like Deepgram JS SDK documentation, can be added here.

File: code/community/1301705198959333384/reduce-latency-deepgram-zoom.rs
use deepgram::DeepgramClient;
use std::env;
use tokio::time::{sleep, Duration};

#[tokio::main]
async fn main() {
    // Set your Deepgram API key as an environment variable
    let api_key = env::var("DEEPGRAM_API_KEY").expect("DEEPGRAM_API_KEY not set");

    // Create a new Deepgram client
    let client = DeepgramClient::new(api_key);

    // URL of the audio stream
    let audio_url = "your-zoom-websocket-url";

    // Additional options can be set here
    let options = deepgram::ListenOptions {
        punctuate: Some(true),
        interim_results: Some(true),
        ..Default::default()
    };

    // Listen to the audio stream and handle incoming transcripts
    let mut stream = client.listen_with_options(audio_url, options).await.expect("Error creating stream");

    while let Some(result) = stream.next().await {
        match result {
            Ok(event) => {
                println!("Transcript: {}", event.transcript);
            }
            Err(err) => {
                eprintln!("An error occurred: {}", err);
            }
        }
    }
}

// For more details, you can refer to the Deepgram Rust SDK documentation.